actors/market/src/lib.rs

```diff
@@ -717,22 +717,38 @@ impl Actor {
                     // part of a sector that is terminating.
                     .ok_or_else(|| actor_error!(illegal_argument, "no state for deal {}", id))?;
 
-                // If a deal is already slashed, don't need to do anything
+                // If a deal is already slashed, there should be no existing state for it
+                // but we process it here for deletion anyway
                 if state.slash_epoch != EPOCH_UNDEFINED {
-                    info!("deal {}, already slashed", id);
-                    continue;
+                    warn!("deal {}, already slashed, terminating now anyway", id);
```

Comment:
I think we should skip process_slashed_deal here, since that's what set the slash epoch. We don't want to settle payments twice.

Reply:
On further thoughts, maybe you're handling the case where a deal was marked for termination before this code went live, but not yet picked up by cron until after. There are some tricky cases here and we need to be crystal clear about what's happening, and reflect that clarity in the comments. Is the case I mentioned the only way to see a slash epoch here? It's not a warning, but something we can simplify later.

Reply:
Your proposal in the FIP to make settlement of a marked-terminated deal an error sounds good to me. Let's simplify.

Reply:
>  Is the case I mentioned the only way to see a slash epoch here? It's not a warning, but something we can simplify later.

Treating marked-for-termination deals as errors will simplify the code in `process_deal_payments`. In here, where it's triggered from `MinerSectorsTermination` it's probably an unreachable error even in the transition phase.

Reply:
I think it can be safely handled as a no-op meaning but probably means something else is wrong if ever triggered.